Terraform improvements for production

AndrewDryga opened the pull request.

AndrewDryga committed:
Bump OTEL metrics collector version

AndrewDryga committed:
Bump Alpine, Elixir, Erlang versions

Fixes CVE in Alpine's openssl lib

AndrewDryga committed:
Minimize chances that multiple gateways can connect with the same ID

It is still possible due to race conditions (connecting on two different nodes at the same time).

AndrewDryga committed:
Write audit log and alert to Slack when somebody connects to prod db

AndrewDryga committed:
Add guide for prod db access to readme

AndrewDryga committed:
Allow using cloud sql proxy and gcloud token to connect to DB

AndrewDryga committed:
Rename ambiguous alert

AndrewDryga committed:
Bump tf vsn

AndrewDryga committed:
Do not report terraform going to prod db

AndrewDryga committed:
Grant permissions to IAM users

AndrewDryga committed:
Add image tag output to staging

AndrewDryga committed:
Also grant IAM users roles on staging

AndrewDryga committed:
Increase admin token ttl

jamilbk commented on terraform/modules/gateway-google-cloud-compute/templates/cloud-init.yaml:
```diff
@@ -157,6 +157,7 @@ runcmd:
   - sudo apt install docker-ce docker-ce-cli containerd.io -y
   - sudo usermod -aG docker $(whoami)
   - sudo systemctl enable docker
+  - sudo echo '{"ipv6":true,"fixed-cidr-v6":"fd00::/80"}' > /etc/docker/daemon.json
```
Was this supposed to be within our unique local prefix or outside of it?

AndrewDryga replied:
@jamilbk This is straight what docker tells us to do and I think @bmanifold did the same for AWS gateways

bmanifold replied:
Yeah, I did follow the instructions on the [Docker IPv6 page](https://docs.docker.com/config/daemon/ipv6/)

The only difference is that I didn't enable it for the default bridge network, but rather created a new network in the docker-compose file.  For reference, here's what I did:

In the `/etc/docker/daemon.json` file
```
{
  "experimental": true,
  "ip6tables": true
}
```

Then in the `docker-compose.yml` file that the gateway is run from:
```


version: '3.8'

services:
  fz_gateway:
    image: us-east1-docker.pkg.dev/firezone-staging/firezone/gateway:1.20231001.0
    ...more-config-here...
    networks:
      - ip6net

networks:
  ip6net:
    enable_ipv6: true
    ipam:
      config:
        - subnet: 2001:0DB8::/112
```

jamilbk reviewed (approved):
LGTM, just have one question

AndrewDryga committed:
Fix shasums

AndrewDryga committed:
Re-request consent screen when Google Workspace is reconnected

This should help with the issue of missing refresh token when the OIDC server wasn't configured properly initially.

AndrewDryga committed:
Compile rust from scratch and bind it's version

The official rust container did not have latest Alpine with fixed CVE
